service/grails-app/services/org/olf/rs/statemodel/StatusService.groovy

```diff
@@ -506,28 +506,29 @@ public class StatusService {
      * @return The state model that is applicable for that request
      */
     public StateModel getStateModel(PatronRequest request) {
-        String settingsKey;
-        String stateModelCode;
-        if (request.isRequester) {
+        String settingsKey = null
+        String stateModelCode = null
+        if (request.isRequester != null) {
+            Boolean isRequester = request.isRequester;
+
             if (request.serviceType?.value == 'copy') {
-                stateModelCode = StateModel.MODEL_NR_REQUESTER;
-            } else if (request?.deliveryMethod?.value == 'url') {
-                stateModelCode = StateModel.MODEL_DIGITAL_RETURNABLE_REQUESTER;
-            } else {
-                settingsKey = SettingsData.SETTING_STATE_MODEL_REQUESTER;
+                settingsKey = isRequester
+                        ? SettingsData.SETTING_STATE_MODEL_REQUESTER_NON_RETURNABLE
+                        : SettingsData.SETTING_STATE_MODEL_RESPONDER_NON_RETURNABLE
+            } else if (request.serviceType?.value == 'loan') {
+                settingsKey = isRequester
+                        ? SettingsData.SETTING_STATE_MODEL_REQUESTER_RETURNABLE
+                        : SettingsData.SETTING_STATE_MODEL_RESPONDER_RETURNABLE
+            } else if (request.deliveryMethod?.value == 'url') {
```

Comment:
This doesn't match the old logic and this branch will never execute. This condition should be under the`loan` condition. I would prefer you keeped the old logic as is:
```
if serviceType == copy {
    non-returnables
} else {
    if deliveryMethod == URL {
      digital-returnables
    } else {
       returnables
    }
}
```